test/python/fe_api/sdpa/test_sdpa_sm80_cutedsl_gate.py

```
@@ -0,0 +1,38 @@
+
+
+@pytest.fixture
+def old_dsl(monkeypatch):
+    monkeypatch.setattr(buffers, "_DSL_STATE", (True, ("nvidia-cutlass-dsl", "4.6.2")))
+
+
+def test_bwd_wrapper_declines_by_version(old_dsl):
+    from cudnn.sdpa.bwd.api_dsl import sdpa_bwd_wrapper_sm80
+
+    t = torch.zeros(1, 1, 8, 16, dtype=torch.float16)
+    with pytest.raises(NotImplementedError, match="found 4.6.2"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)